graph/src/components/server.rs

```diff
@@ -107,3 +112,11 @@ pub trait GraphQLServer {
         port: u16,
     ) -> Result<Box<Future<Item = (), Error = ()> + Send>, Self::ServeError>;
 }
+
+pub trait JsonRpcServer {
+    fn serve(
+        port: u16,
+        provider: Arc<impl SubgraphProvider>,
+        logger: Logger,
+    ) -> Result<Server, io::Error>;
```

Comment:
I agree the direct dependency is bad. I'll try an associated type, though there is no need for a trait because no interaction with the `Server` is required after it starts.